src/state.cc

```diff
@@ -93,10 +93,13 @@ Edge* State::AddEdge(const Rule* rule) {
 }
 
 Node* State::GetNode(StringPiece path, uint64_t slash_bits) {
-  Node* node = LookupNode(path);
-  if (node)
+  if (Node* node = LookupNode(path))
     return node;
-  node = new Node(path.AsString(), slash_bits);
+  return CreateNode(path, slash_bits);
+}
+
+Node* State::CreateNode(StringPiece path, uint64_t slash_bits) {
+  Node* node = new Node(path.AsString(), slash_bits);
```

Comment:
This issue was in the original code, but it looks like this makes an unnecessary string copy due to the definition of the Node::Node() constructor.

What do you think about changing its definition to:

```
Node(std::string path, uint64_t slash_bits)
   : path(std::move(path)), slash_bits(slash_bits) {}
```

to avoid that?
```

Reply:
I made that change in a dedicated preceding commit.